1-js/02-first-steps/15-function-basics/article.md

```diff
@@ -1,40 +1,42 @@
 # Functions
 
-Quite often we need to perform a similar action in many places of the script.
+अक्सर हमें स्क्रिप्ट के कई स्थानों पर इसी तरह की कार्रवाई करने की आवश्यकता होती है।
 
-For example, we need to show a nice-looking message when a visitor logs in, logs out and maybe somewhere else.
+उदाहरण के लिए, जब कोई परिदर्शक इन करता है, लॉग आउट करता है और शायद कहीं और एक अच्छा दिखने वाला संदेश दिखाना चाहता है।
 
-Functions are the main "building blocks" of the program. They allow the code to be called many times without repetition.
+Functions प्रोग्राम के मुख्य "बिल्डिंग ब्लॉक्स" हैं। वे बिना दोहराव के कोड को कई बार कॉल करने की अनुमति देते हैं।
 
-We've already seen examples of built-in functions, like `alert(message)`, `prompt(message, default)` and `confirm(question)`. But we can create functions of our own as well.
+हम पहले से ही बिल्ट-इन function के उदाहरण देख चुके हैं, जैसे `अलर्ट (संदेश)`, `प्रॉम्प्ट (संदेश, डिफ़ॉल्ट)` और `पुष्टि करें (प्रश्न)`। लेकिन हम अपने स्वयं के functions भी बना सकते हैं।
 
 ## Function Declaration
 
-To create a function we can use a *function declaration*.
+## Function घोषणा
 
-It looks like this:
+function बनाने के लिए हम एक _function declaration_ का उपयोग कर सकते हैं।
+
+यह इस तरह दिख रहा है:
 
 ```js
 function showMessage() {
-  alert( 'Hello everyone!' );
+  alert( "सभी को नमस्कार!" );
 }
 ```
 
-The `function` keyword goes first, then goes the *name of the function*, then a list of *parameters* between the parentheses (comma-separated, empty in the example above) and finally the code of the function, also named "the function body", between curly braces.
+`function` कीवर्ड पहले जाता है, फिर _name of the function_ जाता है, फिर कोष्ठक के बीच _parameters_ की एक सूची (उपरोक्त उदाहरण में खाली) और अंत में function का कोड, जिसे "the function body" भी कहा जाता है , घुंघराले ब्रेसिज़ के बीच।
```

Comment:
Why are we using a notation like - _name of the function_? Applies to all the places we are using it.